src/Microsoft.DotNet.Build.Tasks/PackageFiles/VersionTools.targets

```diff
@@ -44,6 +46,51 @@
     </ItemGroup>
   </Target>
 
+  <!--
+    Read a .gitmodules file and generate updaters based on the contents, if GitModulesPath provided.
+    
+    If a submodule has 'branch = ...' configured with a target remote branch to update to and
+    'sourceBuildAutoUpdate = true', this target creates an updater for it.
+    
+    To restrict this target to only create updaters for a specific submodule, set the
+    'TargetSubmodule' property to the name/path of the submodule.
+  -->
+  <Target Name="CreateDefaultSubmoduleUpdateSteps"
+          Condition="'$(GitModulesPath)' != ''">
+    <ReadGitConfigFile File="$(GitModulesPath)">
+      <Output TaskParameter="SubmoduleConfiguration" ItemName="SubmoduleConfiguration" />
+    </ReadGitConfigFile>
+
+    <ItemGroup>
+      <_UpgradableSubmodule Include="@(SubmoduleConfiguration)"
+                            Condition="'%(SubmoduleConfiguration.Branch)' != '' AND
+                                       '%(SubmoduleConfiguration.SourceBuildAutoUpdate)' == 'true'" />
```

Comment:
Is `SourceBuildAutoUpdate` the best name for this property, considering that we are no longer in the `dotnet/source-build` repo, but in BuildTools?

Reply:
Good catch, renaming to `VersionToolsAutoUpdate`.